compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -13218,21 +13218,23 @@ private STNode parseReceiveField() {
                 return STNodeFactory.createSimpleNameReferenceNode(functionKeyword);
             case IDENTIFIER_TOKEN:
                 STNode identifier = parseIdentifier(ParserRuleContext.RECEIVE_FIELD_NAME);
-                return createQualifiedReceiveField(identifier);
+                return createReceiveField(identifier);
             default:
                 recover(peek(), ParserRuleContext.RECEIVE_FIELD);
                 return parseReceiveField();
         }
     }
 
-    private STNode createQualifiedReceiveField(STNode identifier) {
+    private STNode createReceiveField(STNode identifier) {
         if (peek().kind != SyntaxKind.COLON_TOKEN) {
             return STNodeFactory.createSimpleNameReferenceNode(identifier);
         }
 
+        identifier = STNodeFactory.createSimpleNameReferenceNode(identifier);
         STNode colon = parseColon();
         STNode peerWorker = parsePeerWorkerIdentifier();
-        return createQualifiedNameReferenceNode(identifier, colon, peerWorker);
+        peerWorker = STNodeFactory.createSimpleNameReferenceNode(peerWorker);
```

Comment:
Not related to your change. Since we had a QualifiedNameRefenceNode `peerWorker` has to be an `IdentifierToken`. Now that we have a `Token` we can keep the `function` token as it is instead converting it to an identifier. See L13052.  Then it will be consistent with L13218 as well.

https://github.com/ballerina-platform/ballerina-lang/blob/e116c9b89c5d5b73aa6354a5a7c986041ce2b5b9/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java#L13050-L13052